compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt

```diff
@@ -32,15 +43,74 @@ internal actual class ComposeWindow : UIViewController {
 
     private lateinit var layer: ComposeLayer
     private lateinit var content: @Composable () -> Unit
+    private val keyboardVisibilityListener = object : NSObject() {
+        @Suppress("unused")
+        @ObjCAction
+        fun keyboardWillShow(arg: NSNotification) {
+            val keyboardInfo = arg.userInfo!!["UIKeyboardFrameEndUserInfoKey"] as NSValue
+            val keyboardHeight = keyboardInfo.CGRectValue().useContents { size.height }
+            val screenHeight = UIScreen.mainScreen.bounds.useContents { size.height }
+            val focused = layer.getActiveFocusRect()
+            if (focused != null) {
+                val focusedBottom = focused.bottom + getTopLeftOffset().y
+                val hiddenPartOfFocusedElement = focusedBottom + keyboardHeight - screenHeight
+                if (hiddenPartOfFocusedElement > 0) {
+                    // If focused element hidden by keyboard, then change UIView bounds.
+                    // Focused element will be visible
+                    view.setClipsToBounds(true)
+                    val (width, height) = getViewFrameSize()
+                    view.layer.setBounds(
+                        CGRectMake(
+                            x = 0.0,
+                            y = hiddenPartOfFocusedElement,
+                            width = width.toDouble(),
+                            height = height.toDouble()
+                        )
+                    )
+                }
+            }
+        }
+
+        @Suppress("unused")
+        @ObjCAction
+        fun keyboardWillHide(arg: NSNotification) {
+            val (width, height) = getViewFrameSize()
+            view.layer.setBounds(CGRectMake(0.0, 0.0, width.toDouble(), height.toDouble()))
+        }
+
+        @Suppress("unused")
+        @ObjCAction
+        fun keyboardDidHide(arg: NSNotification) {
+            view.setClipsToBounds(false)
+        }
+    }
+
+    init {
+        NSNotificationCenter.defaultCenter.addObserver(
```

Comment:
Shouldn't we call `removeObserver` in something like `dispose`? Or is it definitely weak-referenced, and we don't need to do that?

Reply:
I will check

Reply:
Good suggestion, I moved addObserver to viewWillAppear method. And removeObserver to viewDidDisappear method.
Now subscriptions cancels.